include/clap/factory/draft/preset-discovery.h

```diff
@@ -0,0 +1,310 @@
+   // should be unique within the container file.
+   //
+   // The load_uri is a machine friendly string used to load the preset inside the container via a
+   // the preset-load plug-in extension. The load_uri can also just be the path if that's what the
+   // extension wants but it could also be some other unique id like a database primary key or a
+   // binary offset. It's use is entirely up to the plug-in.
+   void(CLAP_ABI *begin_contained_preset)(
+      const struct clap_preset_discovery_metadata_receiver *receiver,
+      const char                                           *path,
+      const char                                           *load_uri);
+
+   // Adds a plug-in id that this preset can be used with.
+   // plugin_abi: 0 for CLAP
+   void(CLAP_ABI *add_plugin_id)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                 uint32_t                                              plugin_abi,
+                                 const char                                           *plugin_id);
+
+   // Sets the collection to which the preset belongs to.
+   void(CLAP_ABI *set_collection_id)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                     const char *collection_id);
+
+   // Sets the flags, see clap_preset_discovery_flags.
+   // If unset, they are then inherited from the location.
+   void(CLAP_ABI *set_flags)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                             uint32_t                                              flags);
+
+   // Sets the preset name.
+   // The preset name can be infered from the file name, but it is mandatory to set it for
+   // container's preset.
+   void(CLAP_ABI *set_name)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                            const char                                           *name);
+
+   // Adds a creator name for the preset.
+   void(CLAP_ABI *add_creator)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                               const char                                           *creator);
+
+   // Sets a description of the preset.
+   void(CLAP_ABI *set_description)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                   const char *description);
+
+   // Sets the creation time and last modification time of the preset.
+   // The timestamps are in seconds since EPOCH.
+   // If one of the time isn't known, then set it to 0.
+   // If this function is not called, then the indexer may look at the file's creation and
+   // modification time.
+   void(CLAP_ABI *set_timestamps)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                                  uint64_t creation_time,
+                                  uint64_t modification_time);
+
+   // Adds a feature to the preset.
+   // See plugin-features.h and preset-features.h
```

Comment:
Where do I find preset-features.h? Or is it yet to come?

Reply:
preset feature was removed, because it wasn't complete enough
I will update the header, good catch!